lib/util.js

```diff
@@ -603,13 +603,13 @@ function formatValue(ctx, value, recurseTimes) {
   }
 
   if (recurseTimes < 0) {
-    if (isRegExp(value)) {
+    if (isRegExp(value))
       return ctx.stylize(regExpToString.call(value), 'regexp');
-    } else if (Array.isArray(value)) {
+    if (constructor)
+      return ctx.stylize(`[${constructor.name}]`, 'special');
+    if (Array.isArray(value))
```

Comment:
__Note__: this check is redundant as soon as #14881 has landed. Right now Arrays set the constructor to null and the check has to be done again.